homeassistant/components/zha/core/gateway.py

```diff
@@ -615,8 +633,11 @@ async def _async_device_rejoined(self, zha_device):
         zha_device.update_available(True)
 
     async def async_create_zigpy_group(
-        self, name: str, members: list[GroupMember], group_id: int = None
-    ) -> ZhaGroupType:
+        self,
+        name: str,
+        members: list[GroupMember] | None,
+        group_id: int | None = None,
+    ) -> ZHAGroup | None:
```

Comment:
The return value is a noticeable derivation from the previous typing. AFAICT `None` will be returned if the group name already exists.

At the moment this case isn't handled by the websocket endpoint. Not sure though if it's an actual problem in day to day use. The endpoint can always be changed later. We haven't enabled type checking for `zha/api.py` yet, anyway.
https://github.com/home-assistant/core/blob/a91888a7f81e1bf467395891764eab25913a98d3/homeassistant/components/zha/api.py#L406-L407

Reply:
This case shouldn't be possible. I think the only reason that None is there is because of the get call on the dict can return None but if we can get the group by name we should be able to get it by id. Maybe we should change this in the future to raise if for some crazy reason we get None back...